src/vllm_tgis_adapter/tgis_utils/hub.py

```diff
@@ -0,0 +1,274 @@
+import concurrent
+import datetime
+import glob
+import json
+import logging
+import os
+from collections import defaultdict
+from concurrent.futures import ThreadPoolExecutor
+from functools import partial
+from pathlib import Path
+from typing import Dict, List, Optional
```

Comment:
We should be able to use native types if `from __future__ import annotations` at the top of this file


and replacing `Dict` -> `dict`, `List` -> `list` and `Optional[<type>]` -> `<type> | None`